File: app/src/main/java/ru/arhlib/app/afisha/AfishaActivity.java
package ru.arhlib.app.afisha;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.webkit.WebView;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProviders;
import ru.arhlib.app.MyWebViewClient;
import ru.arhlib.app.R;

public class AfishaActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.webview);

        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
        }

        WebView webView = findViewById(R.id.webview);
        webView.setWebViewClient(new MyWebViewClient());
        webView.setBackgroundColor(0);
        String style = "<style>" + getString(R.string.style_css) + "</style>";
        AfishaViewModel viewModel = ViewModelProviders.of(this).get(AfishaViewModel.class);
        viewModel.getPage(6671).observe(this, page -> {
            // null on first start
            if (page != null) {
                String data = style + page.getContent();
                webView.loadDataWithBaseURL("https://arhlib.ru/", data, "text/html; charset=UTF-8", "UTF-8", null);
            }
        });
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.afisha_menu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                return true;
            case R.id.share:
                startActivity(new Intent()
                        .setAction(Intent.ACTION_SEND)
                        .putExtra(Intent.EXTRA_TEXT, getString(R.string.afisha_url))
                        .setType("text/plain"));
                return true;
            case R.id.browser:
                startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.afisha_url))));
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }
}

File: app/src/main/java/ru/arhlib/app/afisha/AfishaActivity.kt
package ru.arhlib.app.afisha

import android.content.Intent
import android.graphics.Color
import android.net.Uri
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.isVisible
import ru.arhlib.app.R
import ru.arhlib.app.browser.WebViewClientWithCustomTabs
import ru.arhlib.app.databinding.AfishaActivityBinding

class AfishaActivity : AppCompatActivity() {
    private val viewModel: AfishaViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = AfishaActivityBinding.inflate(layoutInflater)
        setContentView(binding.root)

        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        binding.webview.webViewClient = WebViewClientWithCustomTabs()
        binding.webview.setBackgroundColor(Color.TRANSPARENT)

        viewModel.afisha.observe(this, { result ->
            if (result is LoadResult.Success) {
                val data = getStyle() + result.data.getRenderedContent()
                binding.webview.loadDataWithBaseURL("https://arhlib.ru/", data, "text/html; charset=UTF-8", "UTF-8", null)
            }

            binding.progressBar.isVisible = result is LoadResult.Loading
            binding.retryBlock.isVisible = result is LoadResult.Error
        })

        binding.retryButton.setOnClickListener {
            viewModel.load()
        }
    }

    private fun getStyle(): String {
        return "<style>" + getString(R.string.style_css) + "</style>";
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.afisha_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            android.R.id.home -> {
                onBackPressed()
                true
            }
            R.id.share -> {
                startActivity(Intent()
                        .setAction(Intent.ACTION_SEND)
                        .putExtra(Intent.EXTRA_TEXT, getString(R.string.afisha_url))
                        .setType("text/plain"))
                true
            }
            R.id.browser -> {
                startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(getString(R.string.afisha_url))))
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }
}
